### PR TITLE
Issue/127

### PR DESCRIPTION
**Pull request by shmuga**

### CODE REVIEW

File: src/Services/Air/AirErrors.js [hunk 1]
```diff
@@ -109,6 +111,8 @@ Object.assign(AirRuntimeError, createErrorsList({
   UnableToSaveBookingWithExtraSegment: 'Unable to save booking with extra segment',
   NoResidualValue: 'Original ticket has no residual value for this specific itinerary. Issue a new ticket using current fares.',
   TicketsNotIssued: 'Host error during ticket retrieve.',
+  CantDetectExchangeReponse: 'Exchange response is unknown.',
+  ExchangeTokenIncorrect: 'Cant parse exchange token. Please resolve it again.',
```
Comment (dchertousov) [inline src/Services/Air/AirErrors.js]: Please, change `Cant` to `Can not` or to `Can\'t`

File: src/Services/Air/transformers/index.js [hunk 2]
```diff
@@ -4,6 +4,7 @@ import setPassengersAge from './set-passengers-age';
 import setHasFareBasisFlag from './set-has-farebasis-flag';
 import setGroupsForSegments from './set-groups-for-segments';
 import addMetaPassengersBooking from './add-meta-passengers-booking';
+import decodeExchangeToken from './decode-exchange-token-and-build';
```
Comment (dchertousov) [inline src/Services/Air/transformers/index.js]: What does `-and-build` mean?
Use it or remove it.

File: src/Services/Air/validators/reservation-locator.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import { AirValidationError } from '../AirErrors';
+
+export default (params) => {
+  if (params.uapi_reservation_locator === undefined
```
Comment (dchertousov) [inline src/Services/Air/validators/reservation-locator.js]: I'm not sure about format. Is it base64-encoded? You may check it with regexp then. If not, that's ok.

File: src/Services/Air/transformers/index.js (hunk 2 above)
Comment (shmuga) [inline src/Services/Air/transformers/index.js]: It means to build xml as result. Should I change it to `build-xml`?

File: src/Services/Air/validators/reservation-locator.js (hunk 3 above)
Comment (shmuga) [inline src/Services/Air/validators/reservation-locator.js]: I'm not sure it's exactly base64, maybe we just should check if it's string and has length more than one?

File: src/Services/Air/transformers/index.js (hunk 2 above)
Comment (dchertousov) [inline src/Services/Air/transformers/index.js]: I mean you use it everywhere as just `decodeExchangeToken`.
I think you could name the file just `decode-exchange-token` then.

File: src/Services/Air/validators/reservation-locator.js (hunk 3 above)
Comment (dchertousov) [inline src/Services/Air/validators/reservation-locator.js]: Aha, that's not token, it's a string from uapi.
Yes, just check for a string here.

File: src/Services/Air/validators/exchange-token.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import { AirValidationError } from '../AirErrors';
+
+export default (params) => {
+  if (params.exchangeToken === undefined
```
Comment (dchertousov) [inline src/Services/Air/validators/exchange-token.js]: Here you should check either for string or base64-string, if you'd like to

**Comment by shmuga** [comment]
Fixes #127

**Pull request merged**